server.go

```diff
@@ -186,5 +214,87 @@ type ServerTripper struct{}
 
 func (t *ServerTripper) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	req.Header.Set("User-Agent", serverUserAgent)
+
+	//TODO: Move this whole block elsewhere
+	if serverWorker != nil {
+		var content []byte
+		if req.Method == "GET" {
+			content = []byte("/task/get/" + strconv.Itoa(config.TrackerProject))
+		} else {
+			//todo: this is retarded and should be moved elsewhere
+			buf, _ := ioutil.ReadAll(req.Body)
+			rdr1 := ioutil.NopCloser(bytes.NewBuffer(buf))
+			rdr2 := ioutil.NopCloser(bytes.NewBuffer(buf))
+
+			content, _ = ioutil.ReadAll(rdr1)
+
+			req.Body = rdr2
+		}
+
+		ts := time.Now().Format(time.RFC1123)
+
+		mac := hmac.New(crypto.SHA256.New, serverWorker.Secret)
+		mac.Write(content)
+		mac.Write([]byte(ts))
+		sig := hex.EncodeToString(mac.Sum(nil))
+
+		req.Header.Add("X-Worker-Id", strconv.Itoa(serverWorker.Id))
+		req.Header.Add("Timestamp", time.Now().Format(time.RFC1123))
+		req.Header.Add("X-Signature", sig)
+
+	}
 	return http.DefaultTransport.RoundTrip(req)
 }
+
+const mimeJSON = "application/json"
+
+// https://github.com/simon987/task_tracker/blob/master/api/models.go
```

Comment:
Maybe it's better if we just import the `api` package instead of duplicating models?

Reply:
Some parts of the json tree are not relevant to the crawler